src/neo/SmartContract/InteropService.Blockchain.cs

```diff
@@ -26,68 +29,72 @@ private static bool Blockchain_GetHeight(ApplicationEngine engine)
 
             private static bool Blockchain_GetBlock(ApplicationEngine engine)
             {
-                ReadOnlySpan<byte> data = engine.CurrentContext.EvaluationStack.Pop().GetSpan();
                 UInt256 hash;
-                if (data.Length <= 5)
-                    hash = Ledger.Blockchain.Singleton.GetBlockHash((uint)new BigInteger(data));
-                else if (data.Length == 32)
+                if (engine.TryPop(out uint height))
+                {
+                    hash = Ledger.Blockchain.Singleton.GetBlockHash(height);
+                }
+                else if (engine.TryPop(out ReadOnlySpan<byte> data))
+                {
+                    if (data.Length != 32) return false;
```

Comment:
This `false` is not "catcheable", if we have plans for make all the syscalls "catcheables", we should review all of them.

Reply:
SYSCALLs can't be catchable.

See https://github.com/neo-project/neo-vm/pull/294#discussion_r396055481